### DISCUSSION TITLE
Can someone help me understand how the Supabase backend works?

### DISCUSSION THREAD

**Discussion by jacob-emery** [supabase/supabase#5173] [Questions] [answered]
Dear Supabase team and dear supporters,

I don't really understand how this backend works. I have never deployed a backend, however I did my research and I grasp the idea how they should work, more or less.

Will the docker image (or in the future kubernetes) use AWS Lambdas (once the Supabase functions are completely integrated, using some sort of adapter)? Will the infrastructure become serverless, as a whole? Or will it be possible to use dedicated servers, so that we won't to have to sacrifice speed (i'm referring to the latency in serverless). (I have no idea if docker will trigger serverless lambda functions, or if the docker will act more like a typical backend).

How does hosting differ when users directly use your backend as a service?

Let's say I would like to work on a project where latency has to be very low, is Supabase the right tool for that?

**Comment by steve-chavez** [comment, marked as answer]
> Or will it be possible to use dedicated servers, so that we won't to have to sacrifice speed
> Let's say I would like to work on a project where latency has to be very low, is Supabase the right tool for that?

Yes, you'll always be able to self-host and use a dedicated server to minimize latency, Supabase key components are open source software.
